### ISSUE TITLE
Improve non-global client support

### ISSUE THREAD

**Issue by kalilistic** [goatcorp/DIPs#20]
Per my understanding, other regions are managed through dalamud forks. Would it be possible (or rather worthwhile) to enable main dalamud to handle these? Would need to coordinate with bluefissure and others.

**Comment by Bluefissure** [comment]
Functionally it's not difficult to support different regions/clients, the major problem is the game patch difference. We're maintaining the same dalamud with additional zh-CHS fonts and some minor differences in signatures, but months earlier. Besides that, all good.
Another concern is plugins compatibility:
- Plugins with Japanese support add ja-JP fonts by default, which doesn't include all zh-CHS characters. Besides that, zh-CHS font range is huge, and may cause a crash when adding multiple zh-CHS fonts (I think this is fixed by enlarging the atlas size in the current dalamud).
- Plugins related to servers/worlds are usually not considering CN worlds and SE doesn't even put the DC info of CN worlds in the game files, so we may need other helper functions to determine the CN DC.

I think it's not possible or appropriate to ask plugin developers to test their plugins at CN client, and as said above, the game update may break plugins and since the CN client is always ~one patch older than the intl client, it requires the plugins repo to maintain multiple revisions of the same plugin when there's a breaking change from version bump.

Regarding XIVLauncher, we have XIVLauncherCN ready for testing, the login logic is different since the API & endpoints & auth process are all different, it's open source at [ottercorp/FFXIVQuickLauncher](https://github.com/ottercorp/FFXIVQuickLauncher) and it works well with the SDO login, differences are:
- XIVLauncherCN doesn't have the static password input field and the login method by static password, under the consideration of account security and weird homemade encryption of SDO, instead, we use the QR code scan (by the mobile auth app 叨鱼 of SDO).
- XIVLauncherCN doesn't support WeGame. WeGame login is something like Steam login, we're not going to dig into that at the moment since Tencent is protecting the WeGame by its own encryption/protection module called TP(Tencent Protection) system. Ottercorp devs don't want to decrypt or even open source that to get sued by Tencent possibly, because it's commonly used in other Tencent games.

**Comment by goaaats** [comment]
I think as much as possible should be upstreamed, but I don't think we should have global support directly out of our repos. It's a lot of work to coordinate and I feel like the other teams are more flexible this way.

Upstreaming abstractions that make global versions easier if the maintenance cost on our side isn't too high, improvements, etc. is very welcome though.

**Comment by Bluefissure** [comment]
> I think as much as possible should be upstreamed, but I don't think we should have global support directly out of our repos. It's a lot of work to coordinate and I feel like the other teams are more flexible this way.
> 
> Upstreaming abstractions that make global versions easier if the maintenance cost on our side isn't too high, improvements, etc. is very welcome though.

I agree with this, some abstractions like https://github.com/goatcorp/FFXIVQuickLauncher/pull/949 will make our CN support a lot easier.
